flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/debezium/reader/SnapshotSplitReader.java

```diff
@@ -328,6 +331,13 @@ public void close() {
             if (statefulTaskContext.getBinaryLogClient() != null) {
                 statefulTaskContext.getBinaryLogClient().disconnect();
             }
+            executor.shutdown();
+            boolean isShutdown =
+                    executor.awaitTermination(
+                            EXECUTOR_AWAIT_TERMINATION_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+            if (!isShutdown) {
+                LOG.warn("The thread executor of SnapshotSplitReader wasn't shutdown properly.");
+            }
```

Comment:
Please check not NULL before call it.
```suggestion
       if (executorService != null) {
                executorService.shutdown();
                if (!executorService.awaitTermination(
                        READER_CLOSE_TIMEOUT, TimeUnit.MILLISECONDS)) {
                    LOG.warn(
                            "Failed to close the snapshot split reader in {} ms.",
                            READER_CLOSE_TIMEOUT);
                }
            }
```